### PR TITLE
add postgres, sequelize

### PR DESCRIPTION
**Pull request by mellyssy**
* refactored structure according to the task 
* added posgres, sequelize

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+const port = config.port;
 const routes: CommonRoutesConfig[] = [];
 
 app.use(express.json());
 
+sequelize.sync().then(() => {
+  console.log("All models were synchronized successfully.");
+});
+
 routes.push(new UsersRoutes(app));
```
Comment (fedaipaca) [inline src/app.ts]: This is kinda disturbing me to be honest. It seems like there is a conversation like this:
- Developer: OK, computer, I want to make a new tyre,
-  Computer: You should give me a car to make a new tyre

Probably, I would create routes and then pass that router to make an app,
```
const app = new App(routes);
app.listen();
```

File: src/controllers/users.controller.ts [hunk 2]
```diff
@@ -5,53 +5,58 @@ import {
...
+    const result = await usersService.updateById(id, data);
     res.status(200).send(result);
   }
 
-  deleteUser(req: express.Request, res: express.Response) {
+  async deleteUser(req: express.Request, res: express.Response) {
     const { id } = req.params;
-    const result = usersService.deleteById(id);
-    res.status(200).send(result);
+    const result = await usersService.deleteById(id);
```
Comment (fedaipaca) [inline src/controllers/users.controller.ts:50]: The purpose of isDeleted field is to be able to make soft deletion like in previous task. You can achieve that by using "paraniod" option when defining sequelize models. https://sequelize.org/docs/v6/core-concepts/paranoid/

File: src/controllers/users.controller.ts [hunk 3]
```diff
@@ -5,53 +5,58 @@ import {
...
   }
 
-  createUser(
+  async createUser(
     req: ValidatedRequest<UserPostRequestSchema>,
     res: express.Response
   ) {
     const data = req.body;
-    const id = usersService.create(data);
+    const id = await usersService.create(data);
```
Comment (fedaipaca) [inline src/controllers/users.controller.ts:59]: My bad, I missed that previous review. The validation for password is not checking "password must contain letters and numbers" condition. alphanum() requires the password to be string value to only contain a-z, A-Z, and 0-9. You can use a regex for that.

File: src/services/users.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { CRUD } from "../common/crud.interface";
+import { UserCreationAttributes } from "../common/common.types";
+import Users from "../data-access/users.dao";
+
+class UserService implements CRUD {
```
Comment (fedaipaca) [inline src/services/users.service.ts]: Good idea to write interface for crud. But keep in mind, the I of SOLID. Interface Segregation: Client should not be forced to use/implement an interface which is not relevant to it. Basically, when you create another model which need crud implementation, have to implement autosuggest method as well. You can separate  autosuggest as another interface, and add getAll method to crud. 
PS: The name of autosuggest and desired implementation in the task is kinda weird to me to be honest.  But that's what it is. Maybe they have a reason for upcoming tasks.

**Review by fedaipaca — APPROVED** [review]
Good structure, easy to follow, easy to read. Thanks for your work.

File: src/app.ts (hunk 1 above)
Comment (mellyssy) [reply src/app.ts]: I couldn't figure out how to implement App class to make it work like in your example, but I have changed routes to work with router and pass them in app through `.use()` now. Hope that makes sense.

File: src/controllers/users.controller.ts (hunk 2 above)
Comment (mellyssy) [reply src/controllers/users.controller.ts:50]: Thank you, I have figured later that it was probably what was meant in the task, so I fixed the code to change `isdeleted` flag that was part of original implementation.

File: src/controllers/users.controller.ts (hunk 3 above)
Comment (mellyssy) [reply src/controllers/users.controller.ts:59]: I see, I misread the initial requirement. Changed validation to regex.

**Pull request merged**